packages/cli/lib/lib/webpack/render-html-plugin.js

```diff
@@ -1,24 +1,50 @@
 const { resolve } = require('path');
-const { existsSync } = require('fs');
+const { existsSync, readFileSync, writeFileSync } = require('fs');
 const HtmlWebpackExcludeAssetsPlugin = require('html-webpack-exclude-assets-plugin');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
 const prerender = require('./prerender');
 const createLoadManifest = require('./create-load-manifest');
 const { warn } = require('../../util');
 const { info } = require('../../util');
-let template = resolve(__dirname, '../../resources/template.html');
+let defaultTemplate = resolve(__dirname, '../../resources/template.html');
+
+function read(path) {
+	return readFileSync(resolve(__dirname, path), 'utf-8');
+}
 
 module.exports = async function(config) {
 	const { cwd, dest, isProd, src } = config;
 	const inProjectTemplatePath = resolve(src, 'template.html');
+	let template = defaultTemplate;
 	if (existsSync(inProjectTemplatePath)) {
 		template = inProjectTemplatePath;
 	}
+
+	template = config.template || template;
+
+	let content = read(template);
+	if (/preact\.headEnd|preact\.bodyEnd/.test(content)) {
+		const headEnd = read('../../resources/head-end.ejs');
+		const bodyEnd = read('../../resources/body-end.ejs');
+		content = content
+			.replace(
+				/<%\s+preact\.title\s+%>/,
+				'<%= htmlWebpackPlugin.options.title %>'
+			)
+			.replace(/<%\s+preact\.headEnd\s+%>/, headEnd)
+			.replace(/<%\s+preact\.bodyEnd\s+%>/, bodyEnd);
+
+		// Unfortunately html-webpack-plugin expects a true file,
+		// so we'll create a temporary one.
+		template = resolve(__dirname, 'template.tmp.ejs');
```

Comment:
Can we make this in `temp` dir? So that this unnecessary detail is abstracted away

Reply:
Can't find any references to a `temp` directory in the current code in `master`. Should we create another directory in the users project folder?

Reply:
I was thinking may be the OS's `temp` dir.

Reply:
https://www.npmjs.com/package/tmp

Reply:
TIL, that node has a native `mkdTemp` function for that :tada: 

https://nodejs.org/api/fs.html#fs_fs_mkdtempsync_prefix_options

Reply:
it may not be entirely ideal to create a new temporary directory for every build 🤔

Reply:
Good call :+1: Updated the PR to always reuse the same directory. Let me know what you think :100: